crates/api_common/src/build_response.rs

```diff
@@ -99,11 +99,18 @@ pub async fn send_local_notifs(
   mentions: Vec<MentionData>,
   post_or_comment_id: PostOrCommentId,
   person: &Person,
-  do_send_email: bool,
+  mut do_send_email: bool,
   context: &LemmyContext,
   local_user_view: Option<&LocalUserView>,
   local_instance_id: InstanceId,
 ) -> LemmyResult<Vec<LocalUserId>> {
+  // Check here if email notifications are disabled by instance, because context is not available
+  // in email crate (so it would require a separate param to pass db pool).
+  let site_view = SiteView::read_local(&mut context.pool()).await?;
+  if site_view.local_site.disable_email_notifications {
+    do_send_email = false;
+  }
```

Comment:
Move this up a level, because that's the place where we're nearly always reading the site_view anyway. Then you can leave this untouched, and alter do_send_email one above.

Also you'll probably be able to do

`do_send_email = !disable_email_notifications` for most of the cases.

Reply:
None of the api methods which call this function read SiteView, so it would get way too verbose.